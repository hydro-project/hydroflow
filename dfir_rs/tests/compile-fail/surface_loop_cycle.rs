fn main() {
    let mut df = dfir_rs::dfir_syntax! {
        loop {
            a = identity() -> identity() -> identity() -> identity();
            a -> a;
        }
    };
    df.run_available();
}
