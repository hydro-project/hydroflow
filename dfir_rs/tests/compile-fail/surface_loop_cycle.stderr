error: Operator forms an illegal cycle within a `loop { ... }` block (1/4).
 --> tests/compile-fail/surface_loop_cycle.rs:4:31
  |
4 |             a = identity() -> identity() -> identity() -> identity();
  |                               ^^^^^^^^^^

error: Operator forms an illegal cycle within a `loop { ... }` block (2/4).
 --> tests/compile-fail/surface_loop_cycle.rs:4:45
  |
4 |             a = identity() -> identity() -> identity() -> identity();
  |                                             ^^^^^^^^^^

error: Operator forms an illegal cycle within a `loop { ... }` block (3/4).
 --> tests/compile-fail/surface_loop_cycle.rs:4:59
  |
4 |             a = identity() -> identity() -> identity() -> identity();
  |                                                           ^^^^^^^^^^

error: Operator forms an illegal cycle within a `loop { ... }` block (4/4).
 --> tests/compile-fail/surface_loop_cycle.rs:4:17
  |
4 |             a = identity() -> identity() -> identity() -> identity();
  |                 ^^^^^^^^^^
